core/domain/takeout_service_test.py

```diff
@@ -876,16 +909,109 @@ def test_export_data_for_full_user_trivial_is_correct(self):
+            entity_type=self.THREAD_ENTITY_TYPE,
+            entity_id=self.THREAD_ENTITY_ID,
+            original_author_id=self.USER_ID_1,
+            status=self.THREAD_STATUS,
+            subject=self.THREAD_SUBJECT,
+            has_suggestion=self.THREAD_HAS_SUGGESTION,
+            summary=self.THREAD_SUMMARY,
+            message_count=self.THREAD_MESSAGE_COUNT
+        )
+        feedback_thread_model.put()
+
+        thread_id = feedback_services.create_thread(
+            self.THREAD_ENTITY_TYPE,
+            self.THREAD_ENTITY_ID,
+            self.USER_ID_1,
+            self.THREAD_SUBJECT,
+            self.MESSAGE_TEXT
+        )
+        feedback_services.create_message(
+            thread_id,
+            self.USER_ID_1,
+            self.THREAD_STATUS,
+            self.THREAD_SUBJECT,
+            self.MESSAGE_TEXT
+        )
+
+        # Retrieve all models for export.
+        all_models = [
+            clazz
+            for clazz in self._get_model_classes()
+            if not clazz.__name__ in self.BASE_CLASSES
+        ]
+
+        # Iterate over models and test export policies.
+        for model in all_models:
+            export_method = model.get_export_method()
+            export_policy = model.get_export_policy()
+            exported_props = [
+                prop
+                for prop in model._properties # pylint: disable=protected-access
+                if (export_policy[prop] ==
+                    base_models.EXPORT_POLICY.EXPORTED)
+            ]
+
+            if export_method == base_models.EXPORT_METHOD.NOT_EXPORTED:
+                self.assertEqual(len(exported_props), 0)
+            elif (export_method ==
+                  base_models.EXPORT_METHOD.SINGLE_UNSHARED_INSTANCE):
+                exported_data = model.export_data(self.USER_ID_1)
+                self.assertEqual(
+                    sorted([
+                        python_utils.UNICODE(key)
+                        for key in exported_data.keys()]),
+                    sorted(exported_props)
+                )
+            elif (export_method ==
+                  base_models.EXPORT_METHOD.SHARED_INSTANCE):
+                self.assertIsNotNone(
+                    model.get_field_name_mapping_to_takeout_keys)
+                exported_data = model.export_data(self.USER_ID_1)
+                field_mapping = model.get_field_name_mapping_to_takeout_keys()
+                self.assertEqual(
+                    sorted(exported_props),
+                    sorted(field_mapping.keys())
+                )
+                self.assertEqual(
+                    sorted(exported_data.keys()),
+                    sorted(field_mapping.values())
+                )
+            elif (export_method ==
+                  base_models.EXPORT_METHOD.MULTIPLE_UNSHARED_INSTANCES):
+                exported_data = model.export_data(self.USER_ID_1)
+
+                # Retrieve the first ID.
```

Comment:
Can you explain this part more? Why just the first, what is the significance of what's going on here, etc.?

Maybe add a verbose docstring with examples.

Reply:
Done

Reply:
Done